transport/tlsfrag/writer.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tlsfrag
+
+import (
+	"bytes"
+	"errors"
+	"io"
+)
+
+// clientHelloFragWriter intercepts the initial TLS Client Hello record and splits it into two TLS records based on the
+// return value of frag function. These fragmented records are then written to the base [io.Writer]. Subsequent packets
+// are not modified and are directly transmitted through the base [io.Writer].
+type clientHelloFragWriter struct {
+	base io.Writer
+	done bool // indicates all splitted rcds have been already written to base
+	frag FragFunc
+
+	buf  *clientHelloBuffer // the buffer containing and parsing a TLS Client Hello record
+	rcds *bytes.Buffer      // the buffer containing splitted records what will be written to base
```

Comment:
I believe this holds only one record?
```suggestion
	record *bytes.Buffer      // the buffer containing splitted records what will be written to base
```

Reply:
It contains two splitted records.